packages/fether-react/src/utils/withAccount.js

```diff
@@ -8,17 +8,15 @@ import { withRouter } from 'react-router-dom';
 import { compose, mapProps } from 'recompose';
 import light from '@parity/light.js-react';
 
-import { transactionCountOf$, withoutLoading } from '@parity/light.js';
+import { transactionCountOf$ } from '@parity/light.js';
 import withAccountsInfo from '../utils/withAccountsInfo';
 
 const WithAccount = compose(
   withRouter,
   withAccountsInfo,
   light({
     transactionCount: props =>
-      transactionCountOf$(props.match.params.accountAddress).pipe(
-        withoutLoading()
-      )
+      transactionCountOf$(props.match.params.accountAddress).pipe()
```

Comment:
Yes, you're correct. After making all the changes I added it back again and it's not showing a blank screen anymore

Reply:
If I don't remove it then if it's syncing and I click an account on the Accounts List page, it takes me to the Account (token list page) and shows a blank screen as shown below:

<img width="363" alt="screen shot 2019-02-07 at 6 32 59 pm" src="https://user-images.githubusercontent.com/6226175/52396764-12b30b80-2b07-11e9-9df7-9b2e79562e17.png">

Whereas if I remove it and I do the same thing, then it shows the following as expected:

<img width="362" alt="screen shot 2019-02-07 at 6 32 03 pm" src="https://user-images.githubusercontent.com/6226175/52396772-1e9ecd80-2b07-11e9-82e4-057b7633b43c.png">

According to the API docs, `withoutLoading()` is for when you don't have a loading screen spinner, but since we've got one we shouldn't use it, otherwise it shows the blank screen instead

Reply:
> According to the API docs, `withoutLoading()` is for when you don't have a loading screen spinner, but since we've got one we shouldn't use it, otherwise it shows the blank screen instead

Incorrect: transactionCount$ is an Observable that fires
- `nonce1, Symbol('LOADING'), nonce2, Symbol('LOADING'),nonce3...` when you remove withoutLoading()
- `nonce1, nonce2, nonce3...` when you add withoutLoading()

If you remove withoutLoading, it might happen that we'll send a transation with the nonce `Symbol('LOADING')`, instead of a correct nonce returned by transactionCount$.

Reply:
Maybe adding a `startWith(undefined)` will work? @axelchalon Do you have any better idea?

Reply:
> Maybe adding a `startWith(undefined)` will work? @axelchalon Do you have any better idea?
> 
> Edit: I think `withAccount` should probably not fetch the transaction count, since we use it in this TokensList page, which doesn't require the nonce. Probably we should create another HOC that only fetches transaction count, but that's for another PR.

when a user tries to send a transaction using TxForm.js, the `handleSubmit` method sets the transaction with `sendStore.setTx({ ...values, chainId, token, transactionCount });`

then when they sign the transaction with `signTransactionWithSignature`, it calls the `getEthereumTx` function in fether-react/src/utils/transaction.js, which creates an instance of EthereumTx with txParams, and we need the following to not be undefined `amount, to, gas, chainId, transactionCount, token.address, token.decimals`.

At the moment in TxForm they can't submit the form until the gas is estimated using the `estimatedTxFee` method and until they enter valid form field inputs, so we're already checking that `amount, gas, gasPrice, to` are not undefined

However we're not checking and throwing an error if `chainId, address, transactionCount, token.address, token.decimals` are undefined.

Reply:
I think we should only verify form values in `validateForm`. If the user fills out the form, then `validateForm` is called. If the form itself is valid but if the transaction count hasn't been fetched yet, `validateForm` will return false-ish, and afaik (correct me if I'm wrong) `validateForm` is not going to be called again once the transaction count is fetched; the user needs to change something in the form to re-trigger `validateForm`. I think an option would be to include `transactionCount`, etc. in the form (initialValues); or alternatively to have another form field `dataFetched` derived from the props, `true` when we have `transactionCount` and the rest, and in `validateForm` we just verify that `dataFetched` is true.